StudentProject/StudentDetails.java

```diff
@@ -48,88 +27,86 @@ public static void main(String[] args) {
                     }
-                    System.out.println("--------------------------------------------------");
+                    System.out.println("----------------------------");
                     break;
-                case 3:
+                case SEARCH_STUDENT:
                     boolean found = false;
                     System.out.println("Please give the student id to search? ");
                     System.out.print("Student id:");
-                    int sno1 = s.nextInt();
+                    int StudentId = s.nextInt();
                     System.out.println("--------------------------------------------------");
                     i = c.iterator();
                     while (i.hasNext()) {
                         StudentData e = i.next();
-                        if (e.getSno() == sno1) {
+                        if (e.getStudentID() == StudentId) {
                             System.out.println(e);
                             found = true;
                         }
                     }
-
                     if (!found) {
                         System.out.println("Record Not Found");
                     }
                     System.out.println("--------------------------------------------------");
                     break;
-
-                case 4:
+                case REMOVE_STUDENT:
                     found = false;
-                    System.out.println("Please provide student id? ");
-                    System.out.print("Student id: ");
-                    sno1 = s.nextInt();
+                    System.out.print("Enter Empno to Delete :");
+                    StudentId = s.nextInt();
                     System.out.println("--------------------------------------------------");
                     i = c.iterator();
                     while (i.hasNext()) {
                         StudentData e = i.next();
-                        if (e.getSno() == sno1) {
+                        if (e.getStudentID() == StudentId) {
                             i.remove();
                             found = true;
                         }
                     }
-
                     if (!found) {
                         System.out.println("Record Not Found");
                     } else {
                         System.out.println("Record is Deleted Successfully...!");
                     }
-
-                    System.out.println("--------------------------------------------------");
+                    System.out.println("----------------------------");
                     break;
-                case 5:
+                case UPDATE_STUDENT:
                     found = false;
                     System.out.println("Which student do you want to excute? ");
                     System.out.print("Student id: ");
-                    sno1 = s.nextInt();
-
+                    StudentId = s.nextInt();
                     ListIterator<StudentData> li = c.listIterator();
```

Comment:
Instead of an Iterator, you can also use a for each loop. for(StudentData currentStudent : c) { ... } or even easier, use streams and filter (https://rit-moodle.inso.tuwien.ac.at/mod/page/view.php?id=662).